setup.py

```diff
@@ -42,12 +30,24 @@ def run_meson_build():
     )
     sysargs = meson_call.split(" ")
     sysargs = [arg for arg in sysargs if arg != ""]
-    meson_main(sysargs)
+    p1 = subprocess.run(sysargs, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    setup_log = os.path.join(staging_dir, "setup.log")
+    with open(setup_log, "wb") as f:
+        f.write(p1.stdout)
```

Comment:
Nice way to cleanup the terminal output and store the logs!
I assume writing the stdout after the process is completed is the standard approach within `subprocess`, but I wonder if there are corner cases where a major failure can lead to the logs to be not written out

Reply:
Yep it should spit out the log before throwing any errors. I did test this locally and it seems to work.